osu.Game/Skinning/ResourcesSkin.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System;
+using osu.Framework.Audio;
+using osu.Framework.Audio.Sample;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.OpenGL.Textures;
+using osu.Framework.Graphics.Textures;
+using osu.Framework.IO.Stores;
+using osu.Framework.Platform;
+using osu.Game.Audio;
+
+namespace osu.Game.Skinning
+{
+    /// <summary>
+    /// An <see cref="ISkin"/> that uses an underlying <see cref="IResourceStore{T}"/> with namespaces for resources retrieval.
+    /// </summary>
+    public class ResourcesSkin : ISkin, IDisposable
```

Comment:
Let's call this `ResourceStoreSkin`. Right now it's a bit hard to tell what the class would be doing (to the point you need to read the xmldoc to understand its purpose).

Reply:
`ResourceStoreBackedSkin` maybe?

Reply:
sure, sounds fine.